### PR TITLE
Disable mDNS broadcasting for the ESP32IOBoard by default.

### PR DESCRIPTION
**Pull request by atanisoft**
By default the ESP32IOBoard should not broadcast over mDNS as this
should only be done for nodes which are acting as a hub. The
ESP32IOBoard can be a hub for other nodes to connect to via the
advertised mDNS service details.

### CODE REVIEW

File: arduino/examples/ESP32IOBoard/ESP32IOBoard.ino [hunk 1]
```diff
@@ -344,7 +350,7 @@ void setup()
         new Esp32HardwareCan("esp32can", CAN_RX_PIN, CAN_TX_PIN));
 #endif // USE_CAN
 
-#if defined(USE_WIFI)
+#if defined(USE_WIFI) && defined(BROADCAST_MDNS)
```
Comment (balazsracz) [inline arduino/examples/ESP32IOBoard/ESP32IOBoard.ino]: I would think that MDNS.begin would be inside this if def?

Comment (atanisoft) [reply arduino/examples/ESP32IOBoard/ESP32IOBoard.ino]: It is outside of this section to allow the node's hostname to be registered for lookup. This is also the first step towards an updated version which will connect the client nodes to a hub.

Comment (balazsracz) [reply arduino/examples/ESP32IOBoard/ESP32IOBoard.ino]: So MDNS.begin exports some other service (in addition to the once in this block)?

Comment (atanisoft) [reply arduino/examples/ESP32IOBoard/ESP32IOBoard.ino]: Yes, it starts the mDNS resolver which will be needed for clients to find the hub nodes that are broadcasting.

**Pull request merged**